src/widget/wfindonmenu.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class WFindOnMenu : public QMenu {
+    Q_OBJECT
+  public:
+    explicit WFindOnMenu(
+            QWidget* parent = nullptr);
+    ~WFindOnMenu() override = default;
+
+    enum class Services {
+        SoundCloud,
+        LastFm,
+        Discogs
+    };
+
+    void openTheBrowser(const QString& serviceUrl,
+            const QString& query);
+
+    void createAllServices(const Track& track);
+
+  signals:
+    void triggerBrowser(
```

Comment:
What's the difference between those 2 arguments with similar names and identical types? Very error prone.

Reply:
maybe webLookupQuerry and porperty, or such. 
You may also consider to use enums to release the menu from implementation details.